### PR TITLE
[DIA-1178] `meta-data` GET request and unit tests implemented

### PR DESCRIPTION
**Pull request by Nevazhnovu**

### CODE REVIEW

File: ConsentViewController/Classes/SPError.swift [hunk 1]
```diff
@@ -256,6 +256,14 @@ import Foundation
     public override var campaignType: SPCampaignType { get { .gdpr } set {} }
 }
 
+@objcMembers public class InvalidMetaDataQueryParamsError: SPError {
+    override public var spCode: String { "sp_metric_invalid_meta_data_query_params" }
+}
+
+@objcMembers public class InvalidMetaDataQueryResponseError: SPError {
```
Comment (andresilveirah) [inline ConsentViewController/Classes/SPError.swift:263]: I believe the correct name should be `InvalidMetaDataResponseError` (there's no _query_ in the response).

File: ConsentViewController/Classes/SourcePointClient/SourcePointClient.swift [hunk 2]
```diff
@@ -451,4 +452,45 @@ extension SourcePointClient {
             })
         }
     }
+
+    func metaDataURLWithParams(
+        env: SPCampaignEnv,
+        accountId: Int,
+        propertyId: Int,
+        metadata: ConsentStatusMetaData
```
Comment (andresilveirah) [inline ConsentViewController/Classes/SourcePointClient/SourcePointClient.swift:460]: @Nevazhnovu the `metadata` of the `/meta-data` request is different from the `metadata` from the `/consent-status` request :D 

I believe you'll need to implement another struct to encapsulate it. Its attributes should be:
```
{
    ccpa?: {
        hasLocalData: boolean
        dateCreated?: string
        uuid?: string
    },
    gdpr?: {
        hasLocalData: boolean
        dateCreated?: string
        uuid?: string
    }
}
```
You can base your new struct on the `ConsentStatusMetaData`.

**Review by andresilveirah — CHANGES_REQUESTED** [review]

**Comment by andresilveirah** [comment]
@Nevazhnovu once you address the comments above and add its specs to `UnmockedSourcepointClientSpec` we should be good to go.

**Comment by carmelo-iriti** [comment]
do you usually create tests in a different MR?

**Comment by andresilveirah** [comment]
> do you usually create tests in a different MR?

@carmelo-iriti no. I think tests are part of the feature/bug fix.

**Comment by Nevazhnovu** [comment]
@andresilveirah code review fixes are pushed, pls review!

**Review by andresilveirah — APPROVED** [review]

**Pull request merged**